include/soci/soci-platform.h

```diff
@@ -8,6 +8,26 @@
 #ifndef SOCI_PLATFORM_H_INCLUDED
 #define SOCI_PLATFORM_H_INCLUDED
 
+// Check some C++11 Features
+#if     defined(_MSC_VER)
+// Bug in Visual Studio: __cplusplus still means C++03
+// https://connect.microsoft.com/VisualStudio/feedback/details/763051/
+#define SOCI_HAVE_NOEXCEPT (_MSC_VER >= 1900)
+#define SOCI_HAVE_LAMBDA   (_MSC_VER >= 1600)
+#elif   defined(__clang__) || defined(__GNUG__)
```

Comment:
On 8 May 2015 at 03:15, VZ <notifications@github.com> wrote:

> In include/soci/soci-platform.h
> <https://github.com/SOCI/soci/pull/333#discussion_r29911015>:
>
> > @@ -8,6 +8,26 @@
> >  #ifndef SOCI_PLATFORM_H_INCLUDED
> >  #define SOCI_PLATFORM_H_INCLUDED
> >
> > +// Check some C++11 Features
> > +#if     defined(_MSC_VER)
> > +// Bug in Visual Studio: __cplusplus still means C++03
> > +// https://connect.microsoft.com/VisualStudio/feedback/details/763051/
> > +#define SOCI_HAVE_NOEXCEPT (_MSC_VER >= 1900)
> > +#define SOCI_HAVE_LAMBDA   (_MSC_VER >= 1600)
> > +#elif   defined(__clang__) || defined(__GNUG__)
>
> I think this could be omitted, we should just test for __cplusplus
> directly, there are other C++11 compilers than gcc/clang (e.g. Intel,
> Oracle).
>

Valid point, indeed, unless the author has got a good reason to avoid
__cplusplus and list compilers explicitly.